File: artist-service/src/main/java/com/gallery/artistservice/infrastructure/entities/ArtistEntity.java
package com.gallery.artistservice.infrastructure.entities;

import com.gallery.artistservice.domain.Artist;
import jakarta.persistence.*;

import java.time.LocalDate;

@Entity
@Table(name = "ARTIST")
public class ArtistEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @Column(nullable = false)
    private String name;

    private LocalDate birthDate;
    private String birthplace;
    private String nationality;
    private String photo;

    public ArtistEntity() {}

    public ArtistEntity(Artist artist) {
        this.id = artist.getId().getId();
        this.name = artist.getName();
        this.birthDate = artist.getBirthDate();
        this.birthplace = artist.getBirthplace();
        this.nationality = artist.getNationality();
        this.photo = artist.getPhoto();
    }

    public Artist toDomain() {
        return new Artist(
                new com.gallery.artistservice.domain.ArtistID(this.id),
                this.name,
                this.birthDate,
                this.birthplace,
                this.nationality,
                this.photo
        );
    }

    // Getters și setters
    public int getId() { return id; }
    public void setId(int id) { this.id = id; }

    public String getName() { return name; }
    public void setName(String name) { this.name = name; }

    public LocalDate getBirthDate() { return birthDate; }
    public void setBirthDate(LocalDate birthDate) { this.birthDate = birthDate; }

    public String getBirthplace() { return birthplace; }
    public void setBirthplace(String birthplace) { this.birthplace = birthplace; }

    public String getNationality() { return nationality; }
    public void setNationality(String nationality) { this.nationality = nationality; }

    public String getPhoto() { return photo; }
    public void setPhoto(String photo) { this.photo = photo; }
}

File: app/ArtistService/Infrastructure/Entities/ArtistEntity.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using ArtistService.Domain;

namespace ArtistService.Infrastructure.Entities
{
    [Table("ARTIST", Schema = "dbo")]
    public class ArtistEntity
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Column("Id")]
        public int Id { get; set; }

        [Column("Name")]
        [Required]
        [StringLength(100)]
        public string Name { get; set; }

        [Column("BirthDate")]
        public DateTime? BirthDate { get; set; }

        [Column("Birthplace")]
        [StringLength(100)]
        public string? Birthplace { get; set; }

        [Column("Nationality")]
        [StringLength(50)]
        public string? Nationality { get; set; }

        [Column("Photo")]
        [StringLength(255)]
        public string? Photo { get; set; }

        public ArtistEntity()
        {
            Id = 0;
            Name = "";
            BirthDate = null;
            Birthplace = "";
            Nationality = "";
            Photo = "";
        }

        public ArtistEntity(Artist artist)
        {
            Id = artist.Id.ArtistId;
            Name = artist.Name;
            BirthDate = artist.BirthDate;
            Birthplace = artist.Birthplace;
            Nationality = artist.Nationality;
            Photo = artist.Photo;
        }

        public Artist ToArtist()
        {
            return new Artist(Id, Name, BirthDate, Birthplace, Nationality, Photo);
        }
    }
}